src/platform/web_app_wayland.cc

```diff
@@ -183,17 +173,17 @@ void WebAppWayland::Init(int width,
           std::unique_ptr<WebAppWaylandWindow>(WebAppWaylandWindow::Take()));
   }
 
-  switch (surface_role) {
+  switch (surface_type) {
     case AglShellSurfaceType::kBackground:
       app_window_->SetAglBackground();
-      surface_role_ = surface_role;
+      surface_type_ = surface_type;
       break;
     case AglShellSurfaceType::kPanel:
       app_window_->SetAglPanel(static_cast<int>(panel_type));
-      surface_role_ = surface_role;
+      surface_type_ = surface_type;
       break;
     default:
-      surface_role_ = AglShellSurfaceType::kNone;
+      surface_type_ = AglShellSurfaceType::kNone;
```

Comment:
So regular applications would just default to kNone? How about adding a new,  specifically for this case, called kDesktop? kNone is like an error rather than default. 

Yeah, so browsing a bit below, we start with kNone for all, which is fine, but I think here a kDesktop should be fine.